pkg/token/cache.go

```diff
@@ -42,6 +43,17 @@ type CacheKey struct {
 	Role              string
 }
 
+// UniqueId returns a unique id of this token,
+// ensuring that the id stays unique with Athenz naming rules.
+// Athenz domain naming rule: "[a-zA-Z0-9_][a-zA-Z0-9_-]*")
+// Athenz role naming rule: "[a-zA-Z0-9_][a-zA-Z0-9_-]*"
+// and therefore delimiter "|" is used to separate domain and role for uniqueness.
+func (k CacheKey) UniqueId(tokenType string) string {
+	d := "|" // delimiter; using not allowed character for domain/role
+	return tokenType + d + k.Domain + d + k.Role + d + strconv.Itoa(k.MaxExpiry) + d + strconv.Itoa(k.MinExpiry) + d + k.ProxyForPrincipal
```

Comment:
join this

Reply:
.join(d)

Reply:
https://github.com/AthenZ/k8s-athenz-sia/pull/89/commits/d1df07f49d6c0a09107f29f8b1a3585031ed75b0